### DISCUSSION TITLE
Failed to parse language config: duplicate field `comment-tokens`

### DISCUSSION THREAD

**Discussion by ofzo** [helix-editor/helix#12171] [Troubleshooting]
there is an error for parse language.toml when start helix, I try to use the default language config , and it works good, I don't know where has an error.

```sh
$ hx .
Failed to parse language config: duplicate field `comment-tokens`
in `language`

Press <ENTER> to continue with default language config
```

the content of my language config is too long, I put it on https://gist.github.com/ofzo/2c631daecad4b21ba9c0b59c4af38cd1

**Comment by the-mikedavis** [comment]
See https://github.com/helix-editor/helix/issues/11666, this is most likely from adding a `comment-tokens` configuration for a language which has a `comment-token` configuration in the `languages.toml` in the root of this repo. (Or vice versa, `languages.toml` could set `comment-tokens` while custom language config sets `comment-token`.)

**Reply by mapsgeek** [reply]
i have the same issue without knowing where's the doublication coming from!! i just copy the `languages.toml` in the repo!
can you please elaborate how to debug this issue?! there's no other custom language configs

**Reply by pascalkuthe** [reply]
You should never copy the config file from the repo. Its inckuded in the binary. Only add what you want to overwrite in your config. The configs are merged automatically.

**Reply by Drakx** [reply]
> See #11666, this is most likely from adding a `comment-tokens` configuration for a language which has a `comment-token` configuration in the `languages.toml` in the root of this repo. (Or vice versa, `languages.toml` could set `comment-tokens` while custom language config sets `comment-token`.)

Using this info I was able to resolve by going through each 'comment-token' and removing one by one until it was solved. Thanks.

**Comment by lenninlc** [comment]
I was able to solve it by narrowing down the 5 languages that I needed. It seems that the issue comes from `comment-tokens` entry somewhere in the `languages.toml` file(https://github.com/helix-editor/helix/blob/master/languages.toml). Hope it helps.

**Comment by rcortezk9** [comment]
I'm getting the same error and from what I'm understanding `comment-tokens` should be changed to `comment-tokens` is this correct or is this still an ongoing issue?

**Comment by Fotkurz** [comment]
Having this issue as well with the `languages.toml` from `master`. 
It would be cool to know which block is the issue in but Idk if this is possible.

**Reply by rcortezk9** [reply]
Thank you for replying. I took the advice to add the languages I was working with and that resolved it.

**Comment by cherry-sigma** [comment]
I somehow figured out the problematic configuration: on line 2059, the configuration of R language.

```toml
# [[language]]
# name = "r"
# scope = "source.r"
# injection-regex = "(r|R)"
# file-types = ["r", "R", { glob = ".Rprofile" }, { glob = "Rprofile.site" }, { glob = ".RHistory" }]
# shebangs = ["r", "R"]
# comment-tokens = ["#", "#'"]
# indent = { tab-width = 2, unit = "  " }
# language-servers = [ "r" ]

# [[grammar]]
# name = "r"
# source = { git = "https://github.com/r-lib/tree-sitter-r", rev = "cc04302e1bff76fa02e129f332f44636813b0c3c" }
```

Got this by binary search manually.

**Reply by cherry-sigma** [reply]
More specifically, the "duplicated field" is related to the `[[language]]` part, and the `[[grammar]]` part works right.

I don't know why either.

**Reply by danilasar** [reply]
Deleting or commenting out these lines didn't helped me Apparently, it not enough, there're other bugs.

**Reply by scapoor** [reply]
In addition to `R` language, there is issue with `gleam` language.
After commenting both in `languages.toml`, it was working. However, why it is not working before commenting is still a mystery.

This is valid for the latest [languages.toml](https://github.com/helix-editor/helix/blob/479c3b55847eae060b69e6f0eda0bd95d93a7fa0/languages.toml) in main repository.
```
diff --git a/languages.toml b/languages.toml
index 8f8b9ac..8a34141 100644
--- a/languages.toml
+++ b/languages.toml
@@ -2087,16 +2087,16 @@ language-servers = [ "solc" ]
 name = "solidity"
 source = { git = "https://github.com/JoranHonig/tree-sitter-solidity", rev = "f7f5251a3f5b1d04f0799b3571b12918af177fc8" }

-[[language]]
-name = "gleam"
-scope = "source.gleam"
-injection-regex = "gleam"
-file-types = ["gleam"]
-roots = ["gleam.toml"]
-comment-tokens = ["//", "///", "////"]
-indent = { tab-width = 2, unit = "  " }
-language-servers = [ "gleam" ]
-auto-format = true
+# [[language]]
+# name = "gleam"
+# scope = "source.gleam"
+# injection-regex = "gleam"
+# file-types = ["gleam"]
+# roots = ["gleam.toml"]
+# comment-tokens = ["//", "///", "////"]
+# indent = { tab-width = 2, unit = "  " }
+# language-servers = [ "gleam" ]
+# auto-format = true

 [[grammar]]
 name = "gleam"
@@ -2138,15 +2138,15 @@ language-servers = [ "robotframework_ls" ]
 name = "robot"
 source = { git = "https://github.com/Hubro/tree-sitter-robot", rev = "322e4cc65754d2b3fdef4f2f8a71e0762e3d13af" }

-[[language]]
-name = "r"
-scope = "source.r"
-injection-regex = "(r|R)"
-file-types = ["r", "R", { glob = ".Rprofile" }, { glob = "Rprofile.site" }, { glob = ".RHistory" }]
-shebangs = ["r", "R"]
-comment-tokens = ["#", "#'"]
-indent = { tab-width = 2, unit = "  " }
-language-servers = [ "r" ]
+# [[language]]
+# name = "r"
+# scope = "source.r"
+# injection-regex = "(r|R)"
+# file-types = ["r", "R", { glob = ".Rprofile" }, { glob = "Rprofile.site" }, { glob = ".RHistory" }]
+# shebangs = ["r", "R"]
+# comment-tokens = ["#", "#'"]
+# indent = { tab-width = 2, unit = "  " }
+# language-servers = [ "r" ]

 [[grammar]]
 name = "r"
```

Additionally, simply replacing the `comment-tokens` with `comment-token` for both `gleam` and `r` fixes this as well.

**Comment by srgfrog** [comment]
Found the error appears in Master version today. Languages.toml changes line 756 from 
`[[language]]`
`name = "c-sharp"`
...
`comment-token = "//"`

to 
`comment-tokens = ["//", "///"]`

I fixed my copy by changing comment-tokens to comment-token
